packages/flutter/lib/src/material/data_table.dart

```diff
@@ -0,0 +1,757 @@
+
+  bool get _debugInteractive => onSelectChanged != null || cells.any((DataCell cell) => cell._debugInteractive);
+}
+
+/// The data for a cell of a [DataTable].
+///
+/// One list of [DataCell] objects must be provided for each [DataRow]
+/// in the [DataTable], in the [new DataRow] constructor's `cells`
+/// argument.
+class DataCell {
+  /// Creates an object to hold the data for a cell in a [DataTable].
+  ///
+  /// The first argument is the widget to show for the cell, typically
+  /// a [Text] or [DropDownButton] widget; this becomes the [widget]
+  /// property and must not be null.
+  ///
+  /// If the cell has no data, then a [Text] widget with placeholder
+  /// text should be provided instead, and then the [placeholder]
+  /// argument should be set to true.
+  const DataCell(this.widget, {
+    this.placeholder: false,
+    this.showEditIcon: false,
+    this.onTap
+  });
+
+  /// The data for the row.
+  ///
+  /// Typically a [Text] widget or a [DropDownButton] widget.
+  ///
+  /// If the cell has no data, then a [Text] widget with placeholder
+  /// text should be provided instead, and [placeholder] should be set
+  /// to true.
+  final Widget widget;
+
+  /// Whether the [widget] is actually a placeholder.
+  ///
+  /// If this is true, the default text style for the cell is changed
+  /// to be appropriate for placeholder text.
+  final bool placeholder;
+
+  /// Whether to show an edit icon at the end of the cell.
+  ///
+  /// This does not make the cell actually editable; the caller must
+  /// implement editing behavior if desired (by hooking [onTap]).
+  ///
+  /// If this is set, [onTap] should also be set, otherwise tapping
+  /// the icon will have no effect.
+  final bool showEditIcon;
+
+  /// Invoked if the cell is tapped.
+  ///
+  /// If non-null, the cell will have an ink well.
+  final VoidCallback onTap;
+
+  bool get _debugInteractive => onTap != null;
+}
+
+/// A material design data table.
+///
+/// Displaying data in a table is expensive, because to lay out the
+/// table all the data must be measured twice, once to negotiate the
+/// dimensions to use for each column, and once to find the rendering
+/// given the results of the negotiation.
+///
+// /// For this reason, if you have a lot of data (say, more than a dozen
+// /// rows with a dozen columns, though the precise limits depend on the
+// /// target device), it is suggested that you use a [DataCard] which
+// /// automatically splits the data into multiple pages.
+// ///
+/// See also:
+///
+///  * [DataColumn]
+///  * [DataRow]
+///  * [DataCell]
+///  * <https://www.google.com/design/spec/components/data-tables.html>
+class DataTable extends StatelessWidget {
+  /// Creates a widget describing a data table.
+  ///
+  /// The [columns] argument must be a list of as many [DataColumn]
+  /// objects as the table is to have columns, ignoring the leading
+  /// checkbox column if any. The table must have at least one column.
```

Comment:
I guess what you mean here is that the table must have at least one column, ignoring the checkbox column. Perhaps it would be simpler to say that the columns' length must be greater than 0.